openflow15/nxt_message_test.go

```diff
@@ -12,3 +21,108 @@ func Test_PacketIn2UnMarshal(t *testing.T) {
 	err := pktIn2.UnmarshalBinary(msgBytes)
 	assert.NoError(t, err)
 }
+
+func TestPacketIn2UnMarshalWithLargePacket(t *testing.T) {
+	for _, tt := range []struct {
+		name          string
+		packetLength  uint16
+		bytesSize     int
+		messageLength uint16
+	}{
+		{"SmallPacket", 64, 96, 96},
+		{"MediumPacket", 1500, 1528, 1528},
+		{"LargePacket", 65530, 65560, 24},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			ethPacket := generateEthernetPacket(tt.packetLength)
+			pktIn2PacketProp := &PacketIn2PropPacket{
+				PropHeader: &PropHeader{
+					Type: uint16(NXPINT_PACKET),
+				},
+				Packet: ethPacket,
+			}
+			pktIn2TableProp := &PacketIn2PropTableID{
+				PropHeader: &PropHeader{
+					Type: uint16(NXPINT_TABLE_ID),
+				},
+				TableID: 1,
+			}
+			pktIn2Msg := NewPacketIn2([]Property{pktIn2PacketProp, pktIn2TableProp})
+			pktIn2Msg.Header.Xid = xID
+			messageBytes, err := pktIn2Msg.MarshalBinary()
+			require.NoError(t, err)
+			assert.Equal(t, tt.bytesSize, len(messageBytes))
+
+			decodedMsg, err := Parse(messageBytes)
+			require.NoError(t, err)
+			decodedVendorHeader, ok := decodedMsg.(*VendorHeader)
+			require.True(t, ok)
+			assert.Equal(t, tt.messageLength, decodedVendorHeader.Header.Length)
+			assert.Equal(t, uint32(NxExperimenterID), decodedVendorHeader.Vendor)
+			assert.Equal(t, uint32(Type_PacketIn2), decodedVendorHeader.ExperimenterType)
+		})
+	}
+}
+
+func TestPacketOutMarshalUnMarshal(t *testing.T) {
```

Comment:
same comments as above for this test function

Reply:
updated.